extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxRecorder.java

```diff
@@ -319,4 +323,28 @@ private EventConsumerInvoker createInvoker(String invokerClassName) {
         }
     }
 
+    private void registerCodecs(Map<Class<?>, Class<?>> codecByClass) {
+        for (Map.Entry<Class<?>, Class<?>> codecEntry : codecByClass.entrySet()) {
+            registerCodec(codecEntry.getKey(), codecEntry.getValue());
+        }
+    }
+
+    private void registerCodec(Class<?> typeToAdd, Class<?> messageCodecClass) {
+        try {
+            if (messageCodecClass.isAssignableFrom(MessageCodec.class)) {
+                MessageCodec messageCodec = (MessageCodec) messageCodecClass.newInstance();
+                registerCodec(typeToAdd, messageCodec);
+            } else {
+                LOGGER.warn(String.format("The codec %s does not inherit from MessageCodec ", messageCodecClass.toString()));
+            }
+        } catch (InstantiationException | IllegalAccessException e) {
+            LOGGER.warn("Cannot instantiate the MessageCodec " + messageCodecClass.toString() + " because " + e.getMessage());
```

Comment:
Should it be a warn or should we throw an error? In any case, I think the stacktrace shouldn't be lost.